impl/src/main/java/jakarta/faces/component/UIData.java

```diff
@@ -16,6 +16,11 @@
 
 package jakarta.faces.component;
 
+import static com.sun.faces.facelets.tag.faces.ComponentSupport.restoreFullDescendantComponentDeltaStates;
```

Comment:
This won't work for 5.0 as UIData won't have access to com.sun.faces packages

Reply:
I suggest you move the necessary code into a private Utils class (or reuse one of the existing ones if possible) in the jakarta.faces.component package.